src/runtime_src/core/edge/sw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -619,15 +644,24 @@ namespace xclcpuemhal2 {
           if(mLogStream.is_open())
             mLogStream << __func__ << " setup instance: " << prev_instanceBaseAddr <<" success "<< success << std::endl;
         }
-        delete []memTopology;
-        memTopology = NULL;
       }
       if (mIsKdsSwEmu)
       {
         mCore = new exec_core;
         mSWSch = new SWScheduler(this);
         mSWSch->init_scheduler_thread();
       }
+
+      //Extract EMULATION_DATA from XCLBIN
+      if (emuData && (emuDataSize > 1)) {
+        std::string emuDataFilePath = binaryDirectory + "/emuDataFile";
+        std::ofstream os(emuDataFilePath);
+        os.write(emuData.get(), emuDataSize);
+        std::cout << "emuDataFilePath : " << emuDataFilePath << std::endl;
+        systemUtil::makeSystemCall(emuDataFilePath, systemUtil::systemOperation::UNZIP, binaryDirectory, boost::lexical_cast<std::string>(__LINE__));
+        systemUtil::makeSystemCall(binaryDirectory, systemUtil::systemOperation::PERMISSIONS, "777", boost::lexical_cast<std::string>(__LINE__));
```

Comment:
No need to use boost::lexical_cast.  Please stick with std::to_string or appropriate std::string constructor.
This is what is causing build error, probably missing include of header, but in any regards don't use boost::lexical_cast, it is long obsolete.

Reply:
@stsoe , your comments addressed and submitted latest changes. pl review and amerge